x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/formatter/TextFormat.java

```diff
@@ -55,11 +55,13 @@ String contentType() {
 
         @Override
         protected Character delimiter() {
+            assert false;
```

Comment:
Don't see an advantage in shutting down the node considering ESQL is a module.
In production this should be disabled and in testing the error signals the problems and lets the rest of the tests gets executed instead of shutting down the VM.

Reply:
Assertions have a number of advantages that apply regardless of whether they trip in a module or in `server`:

1. It's much clearer documentation for readers about the expectations they should have about the code. An assertion here says that calling this method is a definite bug, whereas simply throwing an exception leaves it to the reader to hunt up the call stack to find if the exception is handled anywhere. Similarly, [the assertion I added here](https://github.com/elastic/elasticsearch/blob/5f0161a705d30dc0bce45ffc9fd59edc8c06b4a0/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlQueryResponse.java#L141) documents an invariant which is vitally important for the rest of this method to work correctly, but very much not obvious (I had to ask Nik whether it was even true).
2. There's no risk that someone might try and handle such an exception in future, rather than avoiding the situation that got down to calling this method in the first place.
3. It's not just about the Elasticsearch internal test suite: we might not trip assertions in our tests because of some missing coverage, but if some downstream project manages to hit it then it makes it easy for them to attribute their test failure to a ES bug, and report it to us, rather than working around it on the client side.